model_utilities/train/finetuner.py

```diff
@@ -103,31 +135,26 @@ def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, datase
                     self.__logger.error("Error: unexpected exception during training, exception: %s", str(e))
                     continue
 
-                progress_bar.update()
+                progress_bar.update(batch_size)
 
             avg_training_loss = sum(epoch_training_losses) / len(epoch_training_losses)
             training_losses.append(avg_training_loss)
 
             epoch_validation_losses = []
             with torch.no_grad():
-                for batch in val_batches:
-                    batch = list(batch.items())
-                    (lang1, xx), (lang2, yy) = batch
+                self.__logger.info(f"Epoch {epoch + 1}/{num_epochs}: Starting validation")
+                val_progress_bar = trange(val_batch_num * batch_size)
+                for batch in validation_dataloader:
```

Comment:
This loss calculation code is used two times, for training and for validation. Please move it to a function.

Reply:
im scared

moving this to a different story